ee/modules/110-istio/monitoring/prometheus-rules/controlplane.yaml

```diff
@@ -0,0 +1,83 @@
+                )
+        -
+        # count all istiod pods in non running phase for revision
+        count by (label_istio_io_rev)
+                (
+                  (
+                    kube_pod_status_phase{namespace="d8-istio", pod=~"istiod-.+", phase="Running"}
+                    * on (pod) group_left (label_istio_io_rev) kube_pod_labels{} == 0
+                  )
+                * on (label_istio_io_rev) group_left kube_service_labels{namespace="d8-istio", service="istiod"}
+                )
+        # there are no running control plane pods
+        == 0
+      for: 5m
+      labels:
+        severity_level: "3"
+        tier: cluster
+      annotations:
+        plk_markup_format: "markdown"
+        plk_protocol_version: "1"
+        plk_create_group_if_not_exists__d8_istio_multicluster_remote_api_host_failed: D8IstioGlobalControlplaneDoesntWork,tier=~tier,prometheus=deckhouse,kubernetes=~kubernetes
+        plk_grouped_by__d8_istio_multicluster_remote_api_host_failed: D8IstioGlobalControlplaneDoesntWork,tier=~tier,prometheus=deckhouse,kubernetes=~kubernetes
+        description: |
+          Global istio controlplane `{{$labels.label_istio_io_rev}}` doesn' work.
+          Impact — sidecar injection for Pods with global revision doesn't work, validating webhook for istio resources is absent.
+          ```
+          kubectl get pods -n d8-istio -l istio.io/rev={{$labels.label_istio_io_rev}}
+          ```
+        summary: Global controlplane doesn't work.
+    - alert: D8IstioAdditionalControlplaneDoesntWork
+      expr: |
+        # all_revision_pods - revision_non_running_pods == 0 -> controlplane is dead
+        # count all istiod pods for revision
+        count by (label_istio_io_rev)
+                (
+                  (
+                    kube_pod_labels{namespace="d8-istio", pod=~"istiod-.+"}
+                  )
+                  * on (label_istio_io_rev) group_left kube_service_labels{namespace="d8-istio", service=~"istiod-.+"}
+                  # exclude global revision
+                  unless on (label_istio_io_rev) kube_service_labels{namespace="d8-istio", service="istiod"}
+                )
+        -
+        # count all istiod pods in non running phase for revision
+        count by (label_istio_io_rev)
+                (
+                  (
+                    kube_pod_status_phase{namespace="d8-istio", pod=~"istiod-.+", phase="Running"}
+                    * on (pod) group_left (label_istio_io_rev) kube_pod_labels{} == 0
+                  )
+                * on (label_istio_io_rev) group_left kube_service_labels{namespace="d8-istio", service=~"istiod-.+"}
+                # exclude global revision
+                unless on (label_istio_io_rev) kube_service_labels{namespace="d8-istio", service="istiod"}
+                )
+        == 0
+      for: 5m
+      labels:
+        severity_level: "3"
```

Comment:
```
        severity_level: "3"
```
I think it should be 4. By default, Deckhouse does not send alerts higher than this because the first three levels are meant to wake duty engineers.

On the side of our alertmanager, we remap these levels to be more urgent (with a query like severity = severity - 1) for all alerts. So if it is 4 by default, for a production cluster, it will be 3.

Reply:
fixed!